tracing/tracing.cc

```diff
@@ -161,11 +161,11 @@ void tracing::write_timer_callback() {
 }
 
 future<> tracing::shutdown() {
-    tracing_logger.info("Asked to shut down");
     if (_down) {
-        throw std::logic_error("tracing: shutdown() called for the service that is already down");
+        co_return;
     }
 
+    tracing_logger.info("Asked to shut down");
     write_pending_records();
     _down = true;
     _write_timer.cancel();
```

Comment:
This doesn't work if shutdown is called concurrently, since the second call can return before the first is complete.

I don't mind to merge, since it's overall an improvement, but reentrancy in shutdown() is a recipe for obscure bugs.

Reply:
It's re-entrancy is sort of fuse for aborted start. Todays main looks like this (very simplified)

```
    main() {
        co_await service.start()
        auto stop = defer(co_await service.stop())
        ... // start more services
        auto drain_on_shutdown = defer(co_await drain())
    }

    drain() {
        co_await service.shutdown()
        ... // shutdown more services
    }
```

so in case main() throws before registering drain-on-shutdown, service.stop() would still call service.shutdown(), but if drain-on-shutdown is called, service.main() doesn't shutdown for the 2nd time